update upgrade guide (#157)
bcnmy/docs: docs/Guides/V2Upgrade.md

In the [SDK release](https://github.com/bcnmy/biconomy-client-sdk/releases/tag/r6) (npm version 3.1.1), below API methods have been added to V1 Account API specs (BiconomySmartAccount.ts in accounts package) to help with the upgrades

> - **getUpdateImplementationData( )** : This returns transaction object which can be used in buildUserOp() method for batching with other transactions. To use this method one must enable a validation module. This method can be used while batching the upgrade with other transactions.

Reviewer: ```suggestion
- **getUpdateImplementationData( )** : This returns transaction object which can be used in buildUserOp() method for batching with other transactions. If you do an upgrade by using this transaction payload you must also enable a validation module (get this data by getModuleSetupData()) as part of the Same transaction. These methods can be used while batching the upgrade with other transactions on your new implementation.
```

- **updateImplementationUserOp( )** : This returns the userOp, which can be sent as a **single** action to do the upgrade. 

### Steps to upgrade

1. create a Biconomy account V1 instance.

```tsx
  const biconomyAccount = new BiconomySmartAccount(biconomySmartAccountConfig);
  const biconomySmartAccount = await biconomyAccount.init();
```

2. Create partial userOp using one of the following methods depending on the usecase. 
- If user wants to batch the upgrade transaction with other transactions, `getUpdateImplementationData` method can be used along with `getModuleSetupData`. 

:::info

> Its critical to add the **moduleSetupData** while doing the upgrade when using getUpdateImplementationData to ensure atleast one validation method can be enabled.

Reviewer: ```suggestion
Its critical to add the **moduleSetupData** while doing the upgrade when using getUpdateImplementationData to ensure at least one validation module is enabled. Otherwise your V2 account will be left without any authorization and unusable.
```

:::

  ```tsx
    const txList = [];
    const updrageTransaction = await biconomySmartAccount.getUpdateImplementationData();
    const moduleSetupData = await biconomySmartAccount.getModuleSetupData();
    txList.push(updrageTransaction)
    txList.push(moduleSetupData)
    txList.push(transaction) // any transaction user needs to combine
    const partialUserOp = await biconomySmartAccount.buildUserOp(txList);
  ```

> - If user wants to upgrade the acount explicitly by doing a transaction then `updateImplementationUserOp` method can be used which returns the partial userOp directly. 

Reviewer: ```suggestion
- If user wants to only upgrade the account then `updateImplementationUserOp` method can be used which returns the partial userOp directly. 
```

  ```tsx
    const partialUserOp = await biconomySmartAccount.updateImplementationUserOp() 
    console.log('partial userOp', partialUserOp)
  ```
3. Once we have the partial User operation, we can execute it using following methods.

```tsx

>   // get the paymasterData

Reviewer: add a comment that one can also use ERC20 paymaster here

### Steps to upgrade

> 1. create a Biconomy account V1 instance and execute **`updateImplementationUserOp`** transaction. This includes upgrade transaction and the enabling ECDSA validation module.

Reviewer: there are two things. If they just want to upgrade then it's two actions combined into updateImplementationUserOp

but if something else has to be batched then have to get individual data for getUpdateImplementationData and getEnableSetupModuleData and both MUST be batched along with your own transaction.

```tsx
  const biconomyAccount = new BiconomySmartAccount(biconomySmartAccountConfig);
  const biconomySmartAccount = await biconomyAccount.init();

>   const partialUserOp = await biconomySmartAccount.updateImplementationUserOp() 

Reviewer: we can show both ways either or or. buildUserOp way which forces batching of two (with warning block) or the updateImplementationUserOp way

>   const partialUserOp = await biconomySmartAccount.updateImplementationUserOp() 

Reviewer: will have to also point out that this gives you partial userop without paymasterAndData. if you want to use a paymaster then after this step they have to make a separate call to the paymaster to getPaymasterAndData (this might need some policies to be updated as well like on the account address updateImplementation method and setupAndEnableModule method)
